test/functional/smart_variable_test.rb

```diff
@@ -10,6 +10,7 @@
 
     api_expects(:smart_variables, :update, 'Update the variable') do |par|
       par['smart_variable']['variable'] == 'name2'
+      par['smart_variable']['override_value_order'] = 'fqdn,hostgroup,domain,os'
```

Comment:
There's `&&` missing, to make sure both lines influence result of the test. The latter line is assignment now -> the test will always succeed.